patt/src/main/java/org/example/staticFactorMethod/BMW.java

```diff
@@ -0,0 +1,4 @@
+package org.example.staticFactorMethod;
```

Comment:
This is Factory method pattern.
You need implement static factory method.

https://www.baeldung.com/java-constructors-vs-static-factory-methods